app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package org.oppia.android.app.home.promotedlist
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModel
+import org.oppia.android.R
+import org.oppia.android.app.home.HomeItemViewModel
+import org.oppia.android.app.home.RouteToRecentlyPlayedListener
+import org.oppia.android.app.shim.IntentFactoryShim
+
+/** [ViewModel] promoted story list in [HomeFragment]. */
+class PromotedStoryListViewModel(
+  private val activity: AppCompatActivity,
+  private val internalProfileId: Int,
+  private val intentFactoryShim: IntentFactoryShim,
+  val promotedStoryList: List<PromotedStoryViewModel>
+) :
+  HomeItemViewModel(),
+  RouteToRecentlyPlayedListener {
```

Comment:
Why implement this listener? It seems a bit odd since this is usually implemented by fragments/activities.

Reply:
It looks like needed to override `routeToRecentlyPlayed()` -- if it were moved to the fragment I'm not sure if this view model is able to reference the `HomeFragment` nor does the layout have a reference to the `HomeFragment` if we were to have the layout button directly call `routeToRecentlyPlayed()` (from what I can tell). Might be more reasons that I don't know about (this is the existing implementation!)

Reply:
Ah I see -- yeah I think you're right that it's not being used as a `RouteToRecentlyPlayedListener` other than overriding the function, which could just be called on its own. Removing it doesn't appear to break anything so I'll take it out.